xrpl/utils/tx_parser/orderbook_changes.py

```diff
@@ -0,0 +1,113 @@
+"""Parse order book changes caused by a transaction."""
+from __future__ import annotations
+
+from typing import Any, Dict, List, Union
+
+from xrpl.utils.tx_parser.utils import (
+    group_by_address_order_book,
+    is_valid_metadata,
+    normalize_metadata,
+    normalize_nodes,
+    parse_order_book_changes,
+)
+
+
+class OrderBookChanges:
+    """Parse order book changes from a transactions metadata
+    in a easy-to-read format.
+    """
+
+    # all_orderbook_changes = {'account': ['order book changes']}
+    all_orderbook_changes: Dict[str, List[Dict[str, Any]]] = {}
+
+    def __init__(self: OrderBookChanges, metadata: Any) -> None:
```

Comment:
Does this metadata field re-use the type you defined elsewhere?

Reply:
I defined the parent classes `metadata` as `Any` because I did not wanted to define the huge type hint once again.
So every child class has the exact type hint but the parent classes don't. This also will be improved by using type aliases :)